rllib/policy/eager_tf_policy.py

```diff
@@ -156,54 +148,70 @@ def compute_actions_from_input_dict(
             )
 
         @override(eager_policy_cls)
-        @convert_eager_outputs
-        def _compute_gradients_eager(self, samples: SampleBatch) -> \
-                ModelGradients:
-            """Traced version of EagerTFPolicy's `_compute_gradients_eager`.
+        def learn_on_batch(self, samples):
+            """Traced version of Policy.learn_on_batch."""
+
+            self._check_too_many_retraces()
 
-            Note that `samples` is already zero-padded and has the is_training
-            flag set. We convert this sample batch into tensors here and do the
-            actual computation.
+            # Create a traced version of `self._learn_on_batch_helper`.
+            if self._traced_learn_on_batch_helper is False:
+                self._learn_on_batch_helper = convert_eager_inputs(
```

Comment:
same here. I feel like we should just make self._traced_learn_on_batch_helper as the traced function.

same for self._traced_compute_gradients_helper and self._traced_apply_gradients_helper.

Reply:
See my explanation above.